lib/std/os.zig

```diff
@@ -42,17 +42,15 @@ test {
 }
 
 /// See also `getenv`. Populated by startup code before main().
-/// TODO this is a footgun because the value will be undefined when using `zig build-lib`.
-/// https://github.com/ziglang/zig/issues/4524
-pub var environ: [][*:0]u8 = undefined;
+pub var environ: ?[][*:0]u8 = null;
 
 /// Populated by startup code before main().
 /// Not available on WASI or Windows without libc. See `std.process.argsAlloc`
 /// or `std.process.argsWithAllocator` for a cross-platform alternative.
-pub var argv: [][*:0]u8 = if (builtin.link_libc) undefined else switch (native_os) {
+pub var argv: ?[][*:0]u8 = if (builtin.link_libc) null else switch (native_os) {
```

Comment:
And this breaks ~1.6k usages: https://github.com/search?q=os.argv++path%3A*.zig&type=code&ref=advsearch

That said, IIUC the resolution of #4524 involves the same number of breaks anyhow.